6. Binary Tree/collect_target/[CPP]collect_target_csubhradipta.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file [CPP]collect_target_csubhradipta.cpp
+ * @brief : Check if the input is equal to the sum of any root to leaf path
+ * @details
+ *                  1
+ *                 / \
+ *                /   \
+ *               4     2
+ *               \    / \
+ *                3  5   6
+ * 
+ * The pathsum from root to leaf and input is checked using recursion
+ * From the root, the node value is subtracted from the input value
+ * This is continued until the pointer reaches the leaf nodes
+ * If the input value and leaf node value is same, return true; Else return False
+ * Space Complexity : O(n)
+ * Time Complexity : O(n)
+ * @author [Subhradipta Choudhury](https://github.com/csubhradipta)
+ */
+ 
+#include <bits/stdc++.h> 
+using namespace std; 
+
+struct Node { 
+    int data; 
+    struct Node* left; 
+    struct Node* right; 
+
+    Node(int val) 
+    { 
+        data = val; 
+        left = NULL; 
+        right = NULL; 
+    } 
+};
+
+
+int sum(Node *rt, int n){
+    if (rt == NULL)                 // if such root to leaf path not found
+        return false;
+    else if ((rt->data == n) && (rt->left == NULL) && (rt->right == NULL))      // at last if the input value equals to the leaf node value, return true
+        return true;
+    else

```

Comment:
Just a small piece of advice. When you return something in the function, the function terminates then and there only and it does not go to the next step. So there isn't really any point in giving an else condition. It can all be an if condition.